### ISSUE TITLE
[INFO] Repository is in maintenance mode

### ISSUE THREAD

**Issue by xeffyr** [termux/x11-packages#222]
X11 packages metadata is messed up on Bintray. All packages are going to be re-submitted and won't be available currently.

As temporary solution, pick a mirror by using script `termux-change-repo`, one of:
* By @xeffyr: `deb https://x11.termux-mirror.ml x11 main`
* By Tsinghua University: `deb https://mirrors.tuna.tsinghua.edu.cn/termux/x11-packages/ x11 main`
* By BFSU: `deb https://mirrors.bfsu.edu.cn/termux/x11-packages/ x11 main`

~~Packages [should be available](https://github.com/termux/x11-packages/runs/797313213) in about 2 hours, unless there more failures than I'm aware of.~~

**Comment by Grimler91** [comment]
How did this happen? Something to look out for when updating on the other repos?

**Comment by xeffyr** [comment]
> How did this happen?

Metadata updating job stuck and there now always checksum mismatch between Release and Package files, at least for binary-all.

Repository is configured to do uploads right after all debs are built, so aarch64/arm/i686/x86_64 are uploaded in parallel. This was done to bypass size limit of artifacts archive carried between jobs. Bintray perhaps has a bug or just doesn't like such uploads. Behaviour of parallel uploads is not specified in documentation though.

> Something to look out for when updating on the other repos?

Never upload debs in parallel.

**Comment by xeffyr** [comment]
Observing same issue for termux-packages after `command-not-found` update, but here no parallel uploads were configured.

Stale job https://dl.bintray.com/termux/termux-packages-24/dists/stable/.work_1592831802681/.

Here is what happens during `./build-package.sh -I -a x86_64 ...`:
```
Wrong checksum for https://dl.bintray.com/termux/termux-packages-24/dists/stable/main/binary-x86_64/Packages:
Expected: a47513e6f6827f89b87f9b92ccc55e563495a6cfa3cb04098f76cc5541b8403e
Actual:   b22c1bf5735ccefd504cfcbb3b355da5e58a5808666bf363f4faa9f19ea4f3ed
```

**Comment by Grimler91** [comment]
At least apt upgrade works without problem. Maybe it is a problem on bintray's side of things

**Comment by RalfWerner** [comment]
@xeffyr Could you please _enable_ `Xwayland` again and an another question about the `*-x` packages - did you partially remove/double (imagemagic, mpv ...) what happened in the _**pu**=package update_, with the **_*-x_** request, if the package is missing and why?

**Comment by xeffyr** [comment]
@RalfWerner X11 packages are available partially only, until all packages will be built. I have to rebuild all repository from scratch.

Xwayland won't be available until someone will maintain the related Termux:X11 add-on.

**Comment by RalfWerner** [comment]
> **Xwayland** won't be available until someone will maintain the related Termux:X11 add-on.

Termux only needs **this** package! The rest is in the plugin and installed separately but _**pu**_ failed _**without it!**_ I understand and accept your **WONTFIX** and I **CANTFIX** (knowledge/experience missing) to maintain the _X11 server._ Nevertheless, it would be _"the greatest Termux progress forever"._

Now we have the _"summer of change"_ (@fornwall with `*deb` to `*.apk`). This may also affect the_plugin_ philosophy. Can we discuss X11 in this context and you will resume _Xwayland_ until then?

**Comment by xeffyr** [comment]
@Grimler91 Yes, `apt` works without issues.
```
$ rm -f $PREFIX/var/cache/apt/*.bin $PREFIX/var/lib/apt/lists/*
rm: cannot remove '/data/data/com.termux/files/usr/var/lib/apt/lists/partial': Is a directory
$ apt update
Ign:2 https://dl.bintray.com/grimler/game-packages-24 games InRelease
Ign:3 https://dl.bintray.com/grimler/science-packages-24 science InRelease
Ign:1 https://dl.bintray.com/termux/termux-packages-24 stable InRelease
Get:5 https://dl.bintray.com/grimler/game-packages-24 games Release [5344 B]
Get:6 https://dl.bintray.com/grimler/science-packages-24 science Release [6191 B]
Get:4 https://dl.bintray.com/termux/termux-packages-24 stable Release [8255 B]
Get:7 https://dl.bintray.com/grimler/game-packages-24 games Release.gpg [475 B]
Get:8 https://dl.bintray.com/grimler/science-packages-24 science Release.gpg [475 B]
Get:9 https://dl.bintray.com/termux/termux-packages-24 stable Release.gpg [821 B]
Get:10 https://dl.bintray.com/grimler/game-packages-24 games/stable x86_64 Packages [3979 B]
Get:11 https://dl.bintray.com/grimler/science-packages-24 science/stable x86_64 Packages [8801 B]
Get:12 https://dl.bintray.com/termux/termux-packages-24 stable/main x86_64 Packages [114 kB]
Fetched 149 kB in 2s (59.5 kB/s)
Reading package lists... Done
Building dependency tree       
Reading state information... Done
All packages are up to date.
```

build-package.sh failure probably caused by https://dl.bintray.com/termux/termux-packages-24/dists/stable/.work_1592831802681/ being present. That folder is in Release file and confuses `get_hash_from_file.py`.
```
 701d6031fcfa2c0bfebdad8635e09bf5840cc756528071c942f39c4fc21b0eee            30364 .work_1592831802681/stable/main/binary-all/Packages
 6bef291156f61c593ba5e47fb5b475ea1e9b8ba8f5de27be8a7a3e55385dc3d4             9521 .work_1592831802681/stable/main/binary-all/Packages.bz2
 d693e1a72d0f20fdd28c1cfc87f52ea74e8f4f00c87379588ff4f57e9086b0cf            10788 .work_1592831802681/stable/main/binary-all/Packages.gz
 a47513e6f6827f89b87f9b92ccc55e563495a6cfa3cb04098f76cc5541b8403e           489659 .work_1592831802681/stable/main/binary-x86_64/Packages
 6ba7d2835c066a6ad5448a11be209f62a3e5436a6ee602eed6b8285b32a9d029           115058 .work_1592831802681/stable/main/binary-x86_64/Packages.bz2
 9073696f21cd3509eb2b8f2dc508017241623306af67431cefd635e486a4827c           152557 .work_1592831802681/stable/main/binary-x86_64/Packages.gz
```

```
~/.../termux-packages/scripts $ ./get_hash_from_file.py Release x86_64 main
a47513e6f6827f89b87f9b92ccc55e563495a6cfa3cb04098f76cc5541b8403e
~/.../termux-packages/scripts $ grep a47513e6f6827f89b87f9b92ccc55e563495a6cfa3cb04098f76cc5541b8403e Release
 a47513e6f6827f89b87f9b92ccc55e563495a6cfa3cb04098f76cc5541b8403e           489659 .work_1592831802681/stable/main/binary-x86_64/Packages
```

**Comment by xeffyr** [comment]
Fast builds for termux-packages are now fixed - https://github.com/termux/termux-packages/commit/41806aec0826a314d1db81111ec24e5550b2b001.

**Comment by RalfWerner** [comment]
I deleted the _off-topic_ post above and will add it to the `deb->apk` discussion. That means: you will **not** resume _Xwayland_ until then - Ok?!
With `pkg up` the missing` Xwayland` package is not a problem (will not be up-to-date) until _**pu**_ it can be removed from the `pkg` list. Why not with `imagemagic-x` (removed too)?

**Comment by xeffyr** [comment]
@RalfWerner Install normal `imagemagick`, it now contains all things from imagemagick-x after https://github.com/termux/termux-packages/commit/25d173442628cce0e1f4d5da1ae1e7db92111e12#diff-f8fc5277f0a418e629c8f3bcbadbac25.

I'd like to do same with mpv/mpv-x so we won't have duplicated packages.

**Comment by RalfWerner** [comment]
> I'd like to do same with mpv/mpv-x so we won't have duplicated packages.

That's good! The question was why `pkg in` with `imagemagic-x` **did not** terminate after remove, some weeks ago with `E: unable ..` and `xwayland` **did.** Anyway In the meantime, _**pu**_ also breaks here and I adjust my script and do it again, when [mpv](https://github.com/termux/termux-packages/blob/master/packages/mpv/build.sh) `--enable-x11`.

**Comment by xeffyr** [comment]
> The question was why pkg in with imagemagic-x did not terminate after remove

I don't know what do you mean "did not terminate after remove", but here is what you should see when trying to install `imagemagick-x`:

![Screenshot_20200623-170551_Termux](https://user-images.githubusercontent.com/25881154/85413889-14526780-b574-11ea-9c94-ff6f6153f740.jpg)

Note that mirror still contains old `imagemagick-x`.

**Comment by RalfWerner** [comment]
`apt update` (your current) on my real gigaset device, gives many messages but _nothing_ to `*magick` and the backup` tu.tgz` after that has 28Mb more - no information about incomplete update or too much output. Then I installed/replace your `*95.apk` _Termux_ and **do** the following installation:
```
[~]$ pkg in imagemagick
Reading package lists... Done
Building dependency tree
Reading state information... Done
The following package was automatically installed and is no longer required:
  libtool
Use 'apt autoremove' to remove it.
The following packages will be REMOVED:
  imagemagick-x
The following NEW packages will be installed:
  imagemagick
0 upgraded, 1 newly installed, 1 to remove and 18 not upgraded.
Need to get 7067 kB of archives.
After this operation, 307 kB of additional disk space will be used.
Do you want to continue? [Y/n]
Get:1 https://dl.bintray.com/termux/termux-packages-24 stable/main arm imagemagick arm 7.0.10.20 [7067 kB]
Fetched 7067 kB in 8s (795 kB/s)
(Reading database ... 32113 files and directories currently installed.)
Removing imagemagick-x (7.0.10.0-7) ...
Selecting previously unselected package imagemagick.
(Reading database ... 30887 files and directories currently installed.)
Preparing to unpack .../imagemagick_7.0.10.20_arm.deb ...
Unpacking imagemagick (7.0.10.20) ...
Setting up imagemagick (7.0.10.20) ...
[~]$ type Xwayland
Xwayland is /data/data/com.termux/files/usr/bin/Xwayland
```
... so far so clear! I've also tried to install your `*28.apk` _Termux **without success**_ (not expected but _off-topic_ here)

**Comment by xeffyr** [comment]
Fixed.
